arrow/arrow.py

```diff
@@ -492,7 +492,7 @@ def range(
 
             values = [getattr(current, f) for f in cls._ATTRS]
             current = cls(*values, tzinfo=tzinfo).shift(  # type: ignore[misc]
-                **{frame_relative: relative_steps}
+                check_imaginary=True, **{frame_relative: relative_steps}
```

Comment:
Mind explaining why we want to force usage of check imaginary here rather than making it an argument?

Reply:
To bypass unnecessary checks when performance matters and they know they won't encounter DST transitions.

The default behavior of check_imaginary=True ensures that users get correct and valid results by default. If a user deliberately wants to skip the check for performance reasons or because they're confident it isn't needed, they can do so by setting the parameter to False.

Reply:
Ah my bad, I now see that this was added as true in range and dehumanize shift usages to maintain backwards compatibility right?

Reply:
Yes exactly